gamemodes/terrortown/entities/weapons/weapon_zm_carry.lua

```diff
@@ -734,61 +698,132 @@ if SERVER then
 
 	end
 
-else
-	local draw = draw
+	local ent_diff = vector_origin
+	local ent_diff_time = CurTime()
+	local stand_time = 0
+
+	---
+	-- @realm server
+	function SWEP:Think()
+		BaseClass.Think(self)
+
+		local ctarget = self:GetCarryTarget()
+		local ctype = self:GetCarryType()
+
+		if ctype == CARRY_TYPE_NONE or not self:CheckValidity() then return end
 
-	local PT = LANG.GetParamTranslation
-	local key_params = {primaryfire = Key("+attack", "LEFT MOUSE")}
+		-- If we are too far from our object, force a drop. To avoid doing this
+		-- vector math extremely often (esp. when everyone is carrying something)
+		-- even though the occurrence is very rare, limited to once per
+		-- second. This should be plenty to catch the rare glitcher.
+		if CurTime() > ent_diff_time then
+			ent_diff = self:GetPos() - ctarget:GetPos()
 
+			if ent_diff:Dot(ent_diff) > 40000 then
+				self:Reset()
+
+				return
+			end
+
+			ent_diff_time = CurTime() + 1
+		end
+
+		if CurTime() > stand_time then
+			if PlayerStandsOn(ctarget) then
+				self:Reset()
+
+				return
+			end
+
+			stand_time = CurTime() + 0.1
+		end
+
+		local owner = self:GetOwner()
+
+		if IsValid(self.CarryHack) then
+			self.CarryHack:SetPos(owner:EyePos() + owner:GetAimVector() * 70)
+			self.CarryHack:SetAngles(owner:GetAngles())
+		end
+
+		ctarget:PhysWake()
+	end
+end
+
+
+
+if CLIENT then
 	---
 	-- @ignore
-	function SWEP:DrawHUD()
-		self.BaseClass.DrawHUD(self)
+	function SWEP:Initialize()
+		self:AddTTT2HUDHelp("magneto_stick_help_idle_primary", "magneto_stick_help_idle_secondary")
 
-		if self:GetCanRagPin() and IsValid(self:GetCarriedRag()) then
-			local client = LocalPlayer()
+		return self.BaseClass.Initialize(self)
+	end
+
+	---
+	-- Show relevant strings for the current action.
+	-- @realm shared
+	function SWEP:RefreshTTT2HUDHelp()
+		self.HUDHelp = nil
+
+		local ctarget = self:GetCarryTarget()
+		local ctype = self:GetCarryType()
+		if ctype ~= CARRY_TYPE_NONE and IsValid(ctarget) then
+			if ctype == CARRY_TYPE_RAGDOLL then
+				self:AddTTT2HUDHelp(self:CanPinCurrentRag() and "magneto_stick_help_carry_rag_pin" or "magneto_stick_help_carry_rag_drop", "magneto_stick_help_carry_rag_drop")
+				return
```

Comment:
```suggestion
				self:AddTTT2HUDHelp(
					self:CanPinCurrentRag()
						and "magneto_stick_help_carry_rag_pin"
						or "magneto_stick_help_carry_rag_drop",
					"magneto_stick_help_carry_rag_drop"
				)

				return
```

do the same fotr the other one below